### PR TITLE
Deletedesign

### PR DESCRIPTION
**Pull request by RaviKhandavilli**
Added boiler plate code to support delete design.

### CODE REVIEW

**Review by myungjin — COMMENTED** [review]
Please clean up commits and rebase with a single commit.

File: api/design_api.partials.yml [hunk 1]
```diff
@@ -322,6 +360,54 @@
     tags:
       - designSchemas
 
+  #------------------------------------#
```
Comment (myungjin) [inline api/design_api.partials.yml:363]: This is not your change. It has been merged into the main branch. Please remove this.

File: pkg/openapi/api.go [hunk 2]
```diff
@@ -72,6 +72,7 @@ type DesignCodesApiRouter interface {
 // pass the data to a DesignSchemasApiServicer to perform the required actions, then write the service results to the http response.
 type DesignSchemasApiRouter interface {
 	CreateDesignSchema(http.ResponseWriter, *http.Request)
+	DeleteDesignSchema(http.ResponseWriter, *http.Request)
```
Comment (myungjin) [inline pkg/openapi/api.go:75]: Same here.

File: pkg/openapi/api.go [hunk 3]
```diff
@@ -148,6 +150,7 @@ type DesignCodesApiServicer interface {
 // and updated with the logic required for the API.
 type DesignSchemasApiServicer interface {
 	CreateDesignSchema(context.Context, string, string, DesignSchema) (ImplResponse, error)
+	DeleteDesignSchema(context.Context, string, string, string) (ImplResponse, error)
```
Comment (myungjin) [inline pkg/openapi/api.go:153]: same here

File: pkg/openapi/api_design_schemas.go [hunk 4]
```diff
@@ -52,6 +52,12 @@ func (c *DesignSchemasApiController) Routes() Routes {
 			"/users/{user}/designs/{designId}/schemas",
 			c.CreateDesignSchema,
 		},
+		{
```
Comment (myungjin) [inline pkg/openapi/api_design_schemas.go:55]: Same here

File: pkg/openapi/api_design_schemas.go [hunk 5]
```diff
@@ -154,3 +160,22 @@ func (c *DesignSchemasApiController) UpdateDesignSchema(w http.ResponseWriter, r
 	// If no error, encode the body and the result code
 	EncodeJSONResponse(result.Body, &result.Code, w)
 }
+
+// DeleteDesignSchema - Delete a schema for a given design
+func (c *DesignSchemasApiController) DeleteDesignSchema(w http.ResponseWriter, r *http.Request) {
```
Comment (myungjin) [inline pkg/openapi/api_design_schemas.go:165]: same here

File: pkg/openapi/apiserver/api_design_schemas_service.go [hunk 6]
```diff
@@ -167,3 +168,24 @@ func (s *DesignSchemasApiService) UpdateDesignSchema(ctx context.Context, user s
 
 	return openapi.Response(http.StatusOK, nil), err
 }
+
+func (s *DesignSchemasApiService) DeleteDesignSchema(ctx context.Context, user string, designId string,
```
Comment (myungjin) [inline pkg/openapi/apiserver/api_design_schemas_service.go:172]: same here

File: pkg/openapi/controller/api_design_schemas_service.go [hunk 7]
```diff
@@ -88,3 +89,24 @@ func (s *DesignSchemasApiService) UpdateDesignSchema(ctx context.Context, user s
 
 	return openapi.Response(http.StatusOK, nil), nil
 }
+
+func (s *DesignSchemasApiService) DeleteDesignSchema(ctx context.Context, user string, designId string,
```
Comment (myungjin) [inline pkg/openapi/controller/api_design_schemas_service.go:93]: same here

**Review by myungjin — COMMENTED** [review]
Unnecessary changes were introduced. Please address them.

File: api/design_api.partials.yml (hunk 1 above)
Comment (RaviKhandavilli) [reply api/design_api.partials.yml:363]: done

Comment (RaviKhandavilli) [reply api/design_api.partials.yml]: done

File: pkg/openapi/api.go (hunk 3 above)
Comment (RaviKhandavilli) [reply pkg/openapi/api.go]: done

File: pkg/openapi/api_design_schemas.go (hunk 4 above)
Comment (RaviKhandavilli) [reply pkg/openapi/api_design_schemas.go]: done

File: pkg/openapi/apiserver/api_design_schemas_service.go (hunk 6 above)
Comment (RaviKhandavilli) [reply pkg/openapi/apiserver/api_design_schemas_service.go]: done

File: pkg/openapi/api_design_schemas.go (hunk 5 above)
Comment (RaviKhandavilli) [reply pkg/openapi/api_design_schemas.go]: done

**Review by myungjin — APPROVED** [review]
lgtm

**Pull request merged**